### PR TITLE
Move to the new PluginDependencySpec API

### PR DESCRIPTION
**Pull request by eskatos**
By using `org.gradle.plugin.dsl.PluginDependenciesSpec` from https://github.com/gradle/gradle/pull/2825.

Step for #424

### CODE REVIEW

File: provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt [hunk 1]
```diff
@@ -197,7 +197,7 @@ class KotlinBuildScriptCompiler(
 
         val (lineNumber, compiledScript) = compiledPluginsBlock
         val pluginsBlockClass = classFrom(compiledScript, baseScope.createChild("plugins"))
-        val pluginDependenciesSpec = pluginRequestCollector.createSpec(lineNumber)
+        val pluginDependenciesSpec = pluginRequestCollector.createPluginDependencySpec(lineNumber)
```
Comment (bamboo) [inline provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt]: :o: I understand this is coming from the Gradle API but shouldn't it be called `createPluginDependenciesSpec`?

Comment (eskatos) [reply provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt]: That's right! Let me fix this and I'll get back to you for another pass on this review.

Comment (eskatos) [reply provider/src/main/kotlin/org/gradle/kotlin/dsl/provider/KotlinBuildScriptCompiler.kt]: @bamboo fixed

**Comment by JLLeitschuh** [comment]
Travis CI failed? Should we be concerned?

**Comment by eskatos** [comment]
@JLLeitschuh we should not be, we validated the change before merging, but I agree this is definitely confusing. I opened #504 to track this down

**Comment by JLLeitschuh** [comment]
Thank you!

**Pull request merged**